### ISSUE TITLE
Missing Bijection between `Option[FullName]` and `Option[String]`

### ISSUE THREAD

**Issue by fuzzy-id** [outworkers/util#64]
You can find a test scenario at https://github.com/fuzzy-id/cassandra-tests

I guess it boils down to having some `case class`:
``` scala
case class SimpleFoo( name: Option[String] )
```
Now, doing  `gen[SimpleFoo]` results to an error at compile-time:
```
[error] cassandra-problems/src/test/scala/FooTests.scala:28:16: type mismatch;
[error]  found   : Option[com.outworkers.util.domain.GenerationDomain.FullName]
[error]     (which expands to)  Option[com.outworkers.util.domain.Definitions.FullName]
[error]  required: Option[String]
[error]     val f = gen[SimpleFoo]
[error]                ^
[error] one error found
```

**Comment by alexflav23** [comment]
Hi @fuzzy-id Apologies that I have not gotten round to this yet, I promise I will have a look this weekend, it's a huge challenge finding the time to deal with everything unfortunately, the priority bucketlist is always 1000 pages long :)

**Comment by fuzzy-id** [comment]
@alexflav23 No problem! I would love to look into this myself, but I don't know much about scala yet and don't even know where to start. :(

**Comment by alexflav23** [comment]
@fuzzy-id Sadly this is using whitebox macros and Shapeless, so it may be the wrong end of Scala to start at, but I will find the time to look at it this weekend, hopefully there isn't much work required.

**Comment by alexflav23** [comment]
@fuzzy-id Fixed in 0.40.0, it will be released very shortly.
